control-plane/gateways/deployment_dataplane_container.go

```diff
@@ -27,7 +27,7 @@ const (
 	volumeName                   = "consul-mesh-inject-data"
 )
 
-func consulDataplaneContainer(config GatewayConfig, containerConfig v2beta1.GatewayClassContainerConfig, name, namespace string) (corev1.Container, error) {
+func (b *meshGatewayBuilder) consulDataplaneContainer(containerConfig v2beta1.GatewayClassContainerConfig) (corev1.Container, error) {
```

Comment:
You might be able to get rid of the containerConfig being passed in too, since I believe thats pulled from the GatewayConfig

Reply:
It is, but there is logic in the Deployment func that sets an override for defaults and I wanted to keep that configuration there so it can be done once and then we don't have to worry about the override behavior within the dataplane container func or have any repetition/accidentally forget to do the override elsewhere.

This way, we know the override work is done and can just use it without thinking too hard about it.